src/macros.rs

```diff
@@ -84,8 +85,25 @@ macro_rules! redis_module {
                 ) == raw::Status::Err as c_int { return raw::Status::Err as c_int; }
 
                 $(
-                    if (&$data_type).create_data_type(ctx).is_err() {
-                        return raw::Status::Err as c_int;
+                    {
+                        let type_name = CString::new($data_type::REDIS_TYPE.name).unwrap();
+
+                        let redis_type = unsafe {
```

Comment:
@yihuang don't you think it's better to keep it in native_types.rs?

Reply:
The ``TYPE_METHODS`` table has to be implemented in the user's code, because it's different for each data type.

Reply:
I know but perhaps it's better to pass it as part of RedisType::new 
See the PR I crafted from your PR - https://github.com/RedisLabsModules/redismodule-rs/pull/14/files
https://github.com/RedisLabsModules/redismodule-rs/blob/1d9f5b5607602b2ae388f8c53b59549567be8d41/src/native_types.rs#L19

Reply:
Makes sense.